src/commands/typescript/update.ts

```diff
@@ -33,29 +34,52 @@ export default class Update extends SfdxCommand {
   };
 
   private typescriptPkg: NpmPackage;
+  private repo: SinglePackageRepo | LernaRepo;
+  private packages: Package[];
 
   public async run(): Promise<void> {
     this.typescriptPkg = this.retrieveTsPackage();
     this.validateEsTarget();
     this.validateTsVersion();
 
+    this.repo = (await isMonoRepo())
+      ? await LernaRepo.create({ ux: this.ux })
+      : await SinglePackageRepo.create({ ux: this.ux });
+
+    this.packages = this.getPackages();
+
     this.ux.warn('This is for testing new versions only. To update the version you must go through dev-scripts.');
+
     await this.updateTsVersion();
     await this.updateEsTarget();
 
-    const pkg = await SinglePackageRepo.create({ ux: this.ux });
     try {
-      pkg.install();
-      pkg.build();
-      pkg.test();
+      this.repo.install();
+      this.repo.build();
+      this.repo.test();
     } finally {
       this.ux.log('Reverting unstaged stages');
-      pkg.revertUnstagedChanges();
+      this.repo.revertUnstagedChanges();
     }
   }
 
-  private async updateEsTarget(): Promise<void> {
-    const tsConfigPath = path.resolve('tsconfig.json');
+  private getPackages(): Package[] {
+    return this.repo instanceof LernaRepo ? this.repo.packages : [this.repo.package];
+  }
+
+  private async getPackagePaths(): Promise<string[]> {
```

Comment:
You don't need this method. The path is already exposed on the `Package` class as the `location` property (https://github.com/salesforcecli/plugin-release-management/blob/main/src/package.ts#L54). For example,

```typescript
for (const pkg of this.packages) {
  console.log(pkg.location)
}
```

That being said, I don't think the `location` parameter is add when the `Package` class is instantiated by the `SinglePackageRepo` class. So you might need to update that here: https://github.com/salesforcecli/plugin-release-management/blob/main/src/repository.ts#L493

Reply:
Good, I don't really need the package `location` when dealing with a repo of type `SinglePackageRepo` because when calling the method `updateEsTargetConfig` I pass `'.'` as the package path, but Im instantiating the `Package` class with the current path `pwd().stdout` that way Im able to log the path